src/TuxArchitectStores.js

```diff
@@ -0,0 +1,108 @@
+ArchitectChain.prototype._getRegisterId = function (inputOrStore) {
+  var storeToWaitFor, storeRegistrationId;
+  //if inputOrStore is a string
+  if (typeof inputOrStore === 'string') {
+    //lookup corresponding store in the architecture object
+    storeToWaitFor = architecture[inputOrStore];
+    //if no store maps to this input throw error
+    if (!storeToWaitFor) {
+      throw new Error('store is waiting for an input: "' + inputOrStore + '" that no store outputs.  If this store needs no inputs than only call "itOutputs"');
+    }
+  } else {
+    //if inputOrStore is a store use that
+    storeToWaitFor = inputOrStore;
+  }
+  //return __registerId__ from storeToWaitFor
+  storeRegistrationId = storeToWaitFor.__registerId__;
+  if (storeRegistrationId) {
+    return storeRegistrationId;
+  } else {
+    //if store does not have __registerId__ than it has not been registered to the dispatcher so throw
+    throw new Error('store is waiting for a store that has not been registered to any actions');
+  }
+};
+
+//itNeeds FUNCTION: registers the input(s) and/or store(s) that the store will need to waitFor, returns this architectChain instance to allow for method chaining, updates "and" method to allow chaining via conjuction
+//@param inputOrStore STRING: input string to be checked against the architecture object for the corresponding store [ALTERNATE OBJECT: if a store is passed in than lookup __registerId__ on the passed in store] [ALTERNATE ARRAY: can pass in an array of inputs and/or stores to be registered, method chaining will still function properly in this case]
+ArchitectChain.prototype.itNeeds = function (inputOrStore) {
+  //array of ids to register to the store
+  var waitForIds = [];
+  //if inputOrStore is array
+  if (Array.isArray(inputOrStore)) {
+    var inputOrStoreLength = inputOrStore.length;
+    //loop through inputOrStore array and push register id into waitForIds for each element in array
+    for (var i = 0; i < inputOrStoreLength; i++) {
+      waitForIds.push(this._getRegisterId(inputOrStore[i]));
+    }
+  } else {
+    //if inputOrStore is not an array push its corresponding register id into array
+    waitForIds.push(this._getRegisterId(inputOrStore));
+  }
+  //if the storeToArchitect has already been architected at some point than add waitForIds array on to existing array
+  var tuxArchitecture = this.storeToArchitect.__tuxArchitecture__;
+  if (tuxArchitecture) {
+    tuxArchitecture = tuxArchitecture.concat(waitForIds);
+  } else {
+    //if it has not been architected than just use waitForIds array
+    tuxArchitecture = waitForIds;
+  }
+  //attach architecture array to storeToArchitect at the key of __tuxArchitecture__
+  this.storeToArchitect.__tuxArchitecture__ = tuxArchitecture;
+  //update "and" method so user can chain additional needed inputs with "and"
+  this.and = this.itNeeds;
```

Comment:
Should this be broken out into a separate method? It appears as if this method will need to be called for this reassignment to happen.